src/llscan.cc

```diff
@@ -230,15 +236,53 @@ bool FindInstancesCmd::DoExecute(SBDebugger d, char** cmd,
 
   TypeRecordMap::iterator instance_it =
       llscan_->GetMapsToInstances().find(type_name);
+
   if (instance_it != llscan_->GetMapsToInstances().end()) {
     TypeRecord* t = instance_it->second;
-    for (std::set<uint64_t>::iterator it = t->GetInstances().begin();
-         it != t->GetInstances().end(); ++it) {
+
+    // Update pagination options
+    if (full_cmd != pagination_.command ||
+        inspect_options.output_limit != pagination_.output_limit) {
+      pagination_.total_entries = t->GetInstanceCount();
+      pagination_.command = full_cmd;
+      pagination_.current_page = 0;
+      pagination_.output_limit = inspect_options.output_limit;
+    } else {
+      if (pagination_.output_limit <= 0 ||
+          (pagination_.current_page + 1) * pagination_.output_limit >
+              pagination_.total_entries) {
+        pagination_.current_page = 0;
+      } else {
+        pagination_.current_page++;
+      }
+    }
+
+    int initial_p_offset =
+        (pagination_.current_page * inspect_options.output_limit);
+    int final_p_offset =
+        initial_p_offset +
+        std::min(pagination_.output_limit,
+                 pagination_.total_entries -
+                     pagination_.current_page * pagination_.output_limit);
+    if (final_p_offset <= 0) {
+      final_p_offset = pagination_.total_entries;
+    }
+
+    std::set<uint64_t>::iterator it =
+        pagination_.current_page == 0
+            ? t->GetInstances().begin()
+            : std::next(t->GetInstances().begin(), initial_p_offset);
+    for (; it != t->GetInstances().end() &&
+           it != (std::next(t->GetInstances().begin(), final_p_offset));
+         ++it) {
       Error err;
       v8::Value v8_value(llscan_->v8(), *it);
       std::string res = v8_value.Inspect(&inspect_options, err);
       result.Printf("%s\n", res.c_str());
     }
+    result.Printf("..........\n");
```

Comment:
Suggestion: Only print `"..........\n"` if there are more objects to be printed.